components/insight/SRC/org/openmicroscopy/shoola/util/ui/JXTaskPaneContainerSingle.java

```diff
@@ -197,15 +205,23 @@ public void propertyChange(PropertyChangeEvent evt)
 			c = comp[i];
 			if (c instanceof JXTaskPane) {
 				p = (JXTaskPane) c;
-				if (p != src) {
-					p.setCollapsed(true);
-					p.setSpecial(false);
-					layout.setRow(i, minimumHeight);
-				} else layout.setRow(i, TableLayout.FILL);
+                if (p != src) {
+                    p.setCollapsed(true);
+                    p.setSpecial(false);
+                    GridBagConstraints con = layout.getConstraints(p);
+                    con.fill = GridBagConstraints.HORIZONTAL;
+                    con.weighty = 0;
+                    layout.setConstraints(p, con);
+                } else {
+                    GridBagConstraints con = layout.getConstraints(p);
+                    con.fill = GridBagConstraints.BOTH;
+                    con.weighty = 1;
+                    layout.setConstraints(p, con);
+                }
 			}
 		}
 		src.setSpecial(true);
 		firePropertyChange(SELECTED_TASKPANE_PROPERTY, null, src);
 	}
-
+	
```

Comment:
:) I wonder how that slipped in. Unfortunately the whole formatting is a quite messy, couldn't manage to get my changes in matching the old format and if I reformat the whole file with the right format (i.e. spaces instead of tabs) one can't see the changes in the diff.